workers/default/utils/tasks.py

```diff
@@ -81,16 +81,19 @@ class CleanupDirectoriesTask(BaseTask):
 
     name = "cleanup_directories"
 
-
     def execute_task(self):
-        keep_staging = self.get_param('keep_staging')
-        if (not keep_staging):
-            # delete staging folder if user doesn't want to keep it
-            staging_current_folder = self.get_param('staging_current_folder')
-            user = self.get_param('user_name')
-            staging_path = os.path.join(self.staging_dir, user, staging_current_folder )
-
-            shutil.rmtree(staging_path)
+        mark_solved = self.get_param('mark_solved')
+        staging_current_folder = self.get_param('staging_current_folder')
+        user = self.get_param('user_name')
+        info_file_path = os.path.join(self.staging_dir, user, staging_current_folder, '.info')
+        if mark_solved:
+            f = open(info_file_path, 'w')
+            f.write(
+                'This file marks the parent directory as processed by the iDAI.workbench. Deleting this file will '
+                'unmark the directory in the web interface.')
+            f.close()
+        elif os.path.exists(info_file_path):
+            os.remove(info_file_path)
```

Comment:
Wieso wird die Datei dann gelöscht? Die zwei Zeilen können weg, kann das?

Reply:
@dersmon  Die Datei wird gelöscht, weil es nicht als gelöst markiert ist aber trotzdem das .info File existiert. Was eigentlich ein nicht eintretender Fall sein sollte, so wie ich den Code verstanden habe, aber es schadet auch nicht, das hier zu überprüfen. 

Mein Frage wäre, ob dann das keep_staging feature schon wieder obsolet ist? Mit dieser Lösung werden die ganzen alten Datensätze in der Staging archiviert. Ist das gewollt?

Reply:
ich habe diese auch ersetzt mit der möglichkeit das markieren auszustellen, falls mann die datei als unbearbeitet behalten möchte. könnte mann natürlich auch entfernen.

Reply:
Ich finde das irgendwie unnötig. Ich würde immer alles was erfolgreich bearbeitet wurde auch so markieren...

Reply:
was ist wenn jemand eine datei nochmal bearbeiten will, jedoch tritt dann ein fehler auf. dann finde ich sollte es NICHT als erfolgreich bearbeitet angezeigt werden :)

Reply:
Ok, aber in dem Fall müsste dann die Datei ganz zu Beginn gelöscht werden. In dem von dir beschriebenen Fall würde die ganze Verarbeitungskette ja schon vorher abbrechen.

Reply:
Also die datei bei start eines jobs löschen falls vorhanden ?

Reply:
also im Task Create_Object ?

Reply:
Ja, war mir gerade nicht sicher ob man den Task da zweckentfremdet, aber an sich passt da eine CleanUp Funktion schon noch rein.